### ISSUE TITLE
Athena: Identifiers consisting of only underscores raise parser error

### ISSUE THREAD

**Issue by sbrandtb** [sqlfluff/sqlfluff#3678]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Identifiers consisting of only underscores raise parser error while the queries work in Athena.

### Expected Behaviour

No parser error in all of the examples.

### Observed Behaviour

```
sqlfluff lint --dialect athena underscore*
== [underscore1a.sql] FAIL                                                                                                                                                                                           
L:   1 | P:  10 |  PRS | Line 1, Position 10: Found unparsable section: 'AS _'                                                                                                                                       
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?                                                                                                            
== [underscore1b.sql] FAIL                                                                                                                                                                                           
L:   1 | P:  10 |  PRS | Line 1, Position 10: Found unparsable section: 'AS __'                                                                                                                                      
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?                                                                                                            
== [underscore2a.sql] FAIL                                                                                                                                                                                           
L:   4 | P:   3 |  PRS | Line 4, Position 3: Found unparsable section: 'AS _(a)'                                                                                                                                     
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?                                                                                                            
== [underscore2b.sql] FAIL                                                                                                                                                                                           
L:   4 | P:   3 |  PRS | Line 4, Position 3: Found unparsable section: 'AS __(a)'                                                                                                                                    
WARNING: Parsing errors found and dialect is set to 'athena'. Have you configured your dialect correctly?                                                                                                            
All Finished 📜 🎉!
```

### How to reproduce

```sql
-- underscore1a.sql
SELECT 1 AS _

-- underscore1b.sql
SELECT 1 AS __

-- underscore2a.sql
SELECT a
FROM (
    VALUES ('a'), ('b')
) AS _(a)

-- underscore2b.sql
SELECT a
FROM (
    VALUES ('a'), ('b')
) AS __(a)
```

```
sqlfluff lint --dialect athena underscore*
```

### Dialect

Athena

### Version

```
sqlfluff --version
sqlfluff, version 1.2.1
```

### Configuration

None.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
